src/plugins/usage_collection/server/routes/report_metrics.ts

```diff
@@ -31,11 +31,16 @@
 import { schema } from '@osd/config-schema';
 import { IRouter, ISavedObjectsRepository } from 'opensearch-dashboards/server';
 import { storeReport, reportSchema } from '../report';
+import { BatchReport } from '../types';
+import { ReportSchemaType } from '../report/schema';
 
 export function registerUiMetricRoute(
   router: IRouter,
-  getSavedObjects: () => ISavedObjectsRepository | undefined
+  getSavedObjects: () => ISavedObjectsRepository | undefined,
+  batchingInterval: number
```

Comment:
is this a public api? meaning do other parts of the project use it? if so i would probably make the batchingInterval `number | undefined`. That way we can let developers choose to pass a value. It also might be a little weird to have to pass undefined before a defined value.

Reply:
This is not a public API. It is only being used in one location within usage collector to register the route.